src/Integrator/Integrator.cc

```diff
@@ -389,7 +391,9 @@ void
 Integrator<Dimension>::
 appendPhysicsPackage(Physics<Dimension>& package) {
   if (!havePhysicsPackage(package)) {
+    for (auto* packagePtr: package.preSubPackages()) this->appendPhysicsPackage(*packagePtr);
     mPhysicsPackages.push_back(&package);
+    for (auto* packagePtr: package.postSubPackages()) this->appendPhysicsPackage(*packagePtr);
```

Comment:
It looks like the pre/postSubPackages are only used for the smoothing scale at the moment. A few questions:

- Are there other examples we should consider of things that could become subpackages? 
- Is there a way to do this without the extra complexity of subpackages? If we were missing an appendBoundary instance for one of the subpackages, for instance, that could be difficult to test and find. Although the RK and Voronoi packages are is added directly to the integrator without the extra subpackage indirection, those aren't specific to the hydro package, so maybe this extra machinery makes sense. 
- Do we need to be worried about the case where multiple packages include the same subpackage? If so, we may want to add a check for `if (!havePhysicsPackage(subPackage))`. It seems like the use case is that a subpackage will never be tied to a different package.

Reply:
Those are good questions.  I added these subpackages as a workaround so we could still maintain the user script level ideas of SPH/ASPH, FSISPH/AFSISPH, etc.  I don't know of a better way to maintain that sort of script level backwards compatibility, and I think those concepts of the hydro including the H evolution makes sense from a user perspective.

Another example of something that might become a subpackage is the artificial viscosities.  We could abstract those out of the hydro packages as well, particularly if we add the idea of pair-wise centering data type.

At the moment I don't forsee having subpackages that might be defined by multiple packages, but we can easily check for that since the way the work is the SpheralController builds the ultimate package list by adding in the subpackages from each package.